examples/cryptocurrency-advanced/backend/src/schema.rs

```diff
@@ -24,14 +24,24 @@ use exonum::crypto::{Hash, PublicKey};
 use crate::{wallet::Wallet, INITIAL_BALANCE};
 
 /// Database schema for the cryptocurrency.
+///
+/// Note that the schema is crate-private, but it has a public part.
 #[derive(Debug, FromAccess)]
-pub struct Schema<T: Access> {
-    /// Map of wallet keys to information about the corresponding account.
-    pub wallets: RawProofMapIndex<T::Base, PublicKey, Wallet>,
+pub(crate) struct Schema<T: Access> {
+    /// Public part of the schema.
+    #[from_access(flatten)]
+    pub public: SchemaInterface<T>,
     /// History for specific wallets.
     pub wallet_history: Group<T, PublicKey, ProofListIndex<T::Base, Hash>>,
 }
 
+/// Public part of the cryptocurrency schema.
+#[derive(Debug, FromAccess, RequireArtifact)]
+pub struct SchemaInterface<T: Access> {
```

Comment:
Bikeshedding: IMO `InterfaceSchema` / `PublicSchema` is better, since `XXXInterface` is usually associated with some description abstracted over a hidden implementation.

Reply:
What about `SchemaAccess` ??? 🤔

Reply:
We already have `Access` as an abstraction, and it will cause even more ambiguity, as for me.

Reply:
One of reasons for using `*Interface` is that `*Schema` is processed specially by `#[derive(FromAccess)]`. I'm thinking about removing special processing and using `Schema` for public part of schema and `SchemaImpl` for the full / private part. What do you think about it?

Reply:
Sounds good.